cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -286,28 +293,26 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
 
 	_, _ = logger.Debugf("List Teams  Request :%+v", currentModel)
 
-	// Validate required fields in the request
-	if modelValidation := validateModel(ListRequiredFields, currentModel); modelValidation != nil {
-		return *modelValidation, errors.New("required field not found")
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(util.DefaultProfile)
 	}
 
-	// Create mongo DB client
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf(constants.ErrorCreateMongoClient, err)
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), err
+	client, handlerError := util.NewMongoDBClient(req, currentModel.Profile)
+	if handlerError != nil {
+		_, _ = logger.Warnf("CreateMongoDBClient error: %v", handlerError)
```

Comment:
NewMongoDBClient method already logs errors, is this still needed? if yes, can we move it to NewMongoDBClient instead?

Reply:
If you're updating, can you also consider moving the profile update into the helper method. We have a lot of repeated code here that can be handled by the client.